src/Orckestra.Composer.Search/Factory/ProductSearchViewModelFactory.cs

```diff
@@ -196,5 +209,21 @@ public virtual void MapProductSearchViewModelPricing(ProductSearchViewModel prod
             productSearchVm.IsOnSale = pricing.IsOnSale;
             productSearchVm.PriceListId = pricing.PriceListId;
         }
+
+        // NOTE: when fetching data for products from OCC APIs, make sure to query data in batches for optimal performance
+        // https://docs.orckestra.com/developer-documentation/platform-performance/batch-api-requests
+        public virtual async Task<IList<ProductSearchViewModel>> EnrichAppendProductSearchViewModels(IList<(ProductSearchViewModel, ProductDocument)> productSearchResultList, SearchCriteria criteria)
+        {
+            var _productSettings = await ProductSettings.GetProductSettings(ComposerContext.Scope, ComposerContext.CultureInfo).ConfigureAwait(false);
```

Comment:
To match existing naming convention, the variable name should be "productSettings" 
The underscore prefix is for class fields only